### PR TITLE
Adjust ESLint config and fix errors

### PR DESCRIPTION
**Pull request by nl0**

### CODE REVIEW

**Comment by fiskus** [comment]
I can't approve it right now, because diff has mixed cosmetic changes like `<React.Fragment> -> <>` and logic changes (dependency list for hooks). And the latter needs to be tested

**Comment by fiskus** [comment]
To be clear: I disagree with the extensive use of inline `// eslint-ignore-line` and extending Airbnb's eslint config. So, I'm ok with using these, but it's not our shared decision, it's your personal.

File: catalog/app/utils/APIConnector.js [hunk 1]
```diff
@@ -276,19 +276,19 @@ function* apiSaga({ fetch, base = '', middleware = [] }) {
...
+      try {
+        const result = yield call(execRequest, opts)
+        yield put(response(result, opts))
+        yield call(resolve, result)
+      } catch (e) {
+        yield put(response(e, opts))
+        yield call(reject, e)
+      }
+    },
+  )
```
Comment (fiskus) [inline catalog/app/utils/APIConnector.js:291]: I've called `npm run lint` and task have failed on this file with `prettier/prettier` error

File: catalog/app/components/TalkToUs/TalkToUs.js [hunk 2]
```diff
@@ -72,16 +72,19 @@ export function TalkToUsProvider({ children }) {
...
 export function useTalkToUs(extra) {
-  const show = React.useContext(Ctx)
-  const bound = React.useCallback(() => show(extra), [show])
-  return extra ? bound : show
+  const ref = React.useRef({
+    bound: (localExtra) => ref.current.show(ref.current.extra || localExtra),
+  })
+  ref.current.show = React.useContext(Ctx)
+  ref.current.extra = extra
+  return ref.current.bound
```
Comment (fiskus) [inline catalog/app/components/TalkToUs/TalkToUs.js:87]: Where can I find this page?

File: catalog/app/utils/APIConnector.js (hunk 1 above)
Comment (nl0) [reply catalog/app/utils/APIConnector.js]: there was a newer version of prettier in my node_modules. fixed now

File: catalog/app/components/TalkToUs/TalkToUs.js (hunk 2 above)
Comment (nl0) [reply catalog/app/components/TalkToUs/TalkToUs.js:87]: on the landing page of the website (mode === 'MARKETING') in the following places:
- `app/website/pages/Landing/Pricing`: pricing grid "talk to us" and "contact" btns
- `app/website/pages/Landing/Showcase`: hero, "book a demo" btn
- `app/website/pages/Landing/StickyFooter`: sticky footer "talk to us" btn

Comment (nl0) [reply catalog/app/components/TalkToUs/TalkToUs.js:87]: (it's not a page, just a function opening the calendar popup)

Comment (fiskus) [reply catalog/app/components/TalkToUs/TalkToUs.js:87]: Is it deployed somewhere already?

Comment (fiskus) [reply catalog/app/components/TalkToUs/TalkToUs.js:87]: > it's not a page
dude

**Review by fiskus — APPROVED** [review]
I didn't find any mistyping or broken logic in code, but I can't help you with manual testing, because I'm not so familiar with the project.

I suggest to merge it on Monday, or after open.quiltdata deploy

File: catalog/app/components/TalkToUs/TalkToUs.js (hunk 2 above)
Comment (nl0) [reply catalog/app/components/TalkToUs/TalkToUs.js:87]: > Is it deployed somewhere already?

nope, but you can set `"mode": "MARKETING"` in config.json to test it locally

Comment (fiskus) [reply catalog/app/components/TalkToUs/TalkToUs.js:87]: If it isn't deployed to production, it looks like dead code for me. I don't want to test dead code :)

**Comment by fiskus** [comment]
>> grep -r 'eslint-disable' | wc -l

> 232

232 inline tweaks of eslint config. I consider it as extensive use

----

I spent tens of hours in meetings discussing code style. I want to solve real problems, not arguing what is better `class Comp { static displayName }` or `Comp.displayName`. Everyone has some personal preferences, and we can't create a style guide that we like both.

File: catalog/app/components/TalkToUs/TalkToUs.js (hunk 2 above)
Comment (nl0) [reply catalog/app/components/TalkToUs/TalkToUs.js:87]: oh, i see what you mean, it's deployed on https://quiltdata.com. i thought you were asking is this particular revision deployed somewhere

**Comment by nl0** [comment]
> 232 inline tweaks of eslint config. I consider it as extensive use

this number doesnt mean anything by itself, you can dig deeper and see what each individual override achieves (console-related stuff being ~100 out of 200+ cases you mentioned, which is a known issue).

in most cases the overrides are there bc it's impossible or very inconvenient to achieve what we need completely lint-free in terms of our config, and i believe it's ok. what's the actual problem with this?

> class Comp { static displayName } or Comp.displayName

that rule is just outdated, see the reasoning behind it in [airbnb config](https://github.com/airbnb/javascript/blob/d3c7b84d9e513fc1388de28775574ed614ee33e6/packages/eslint-config-airbnb/rules/react.js#L490)

> Everyone has some personal preferences, and we can't create a style guide that we like both.

true, but sometimes personal preference isnt relevant bc some things are just plain dumb or outdated or used out of context.

and btw it's not that hard to decide on some specifics of style and capture that in configs and some written styleguide while there's only two people working on the codebase. i'm not too religious about most of the stuff, i just dont like following the rules dogmatically and not changing them when the situation calls for that.

**Comment by nl0** [comment]
> 232 inline tweaks of eslint config

down to 219 :)

**Pull request merged**